webassembly/app.html

```diff
@@ -255,10 +278,43 @@ <h1 class="title">F3D Web</h1>
                 openFile(filename);
               });
             })
+            .catch(function (error) {
+              console.log('error caught during fetch', error)
+              alert('Error occurred while fetching the model at provided url (passed as url-param). \n\nThis might be a CORS issue (you can avoid it by using a Allow-CORS plugin) because server does not allow cross-origin requests, or the url to the file is wrong. \n\nComplete error message: ' + error.message)
+            });
           } else {
             // load the file located in the virtual filesystem
             openFile('f3d.vtp');
           }
+
+          // Parse other options like #axis=false&grid=false&fxaa=false&tone=false&ssao=false&ambient=false
+          const options = Module.engineInstance.getOptions()
+          for (let [id, option] of idOptionMappings) {
+            // Set f3d option parameter values
+            const parsed_param_value = params.get(id);
+            if (parsed_param_value && parsed_param_value.toLowerCase() === 'false') {
+              // we can use toggle, set_string, set_integer, set_color
+              // sad thing is toggle requires knowing the param state beforehand, rather than setting it explicitly
+              options.toggle(option)
```

Comment:
As you already figured out, the JS bindings are not complete enough yet to achieve what you're attempting here.
https://github.com/f3d-app/f3d/blob/f3d7627047a8d584d0d2cd469d1bc616ffaca539/webassembly/F3DEmscriptenBindings.cxx#L80-L84

You could use the value of the UI switches to guess the current values and decide whether to toggle but that'd be sketchy as there is no proper guarantee that they'll be in sync.

At the very least `set_boolean` would need to be added, but ideally the bindings should expose the newer `setAsString` method that would let you do `options.set_as_string(option, parsed_param_value)` directly (passing a `"false"`/`"true"` string and having the parsing done internally, as opposed to converting from string to bool in the JS and calling `.set_boolean`).

Reply:
The current implementation takes for granted that: 
 - every DOM `input` is `checked` when page loads, see [this code block](https://github.com/f3d-app/f3d/blob/f3d7627047a8d584d0d2cd469d1bc616ffaca539/webassembly/app.html#L54-L76) 
 - the corresponding f3d `settings.options` are toggled on, see this [code block](https://github.com/f3d-app/f3d/blob/f3d7627047a8d584d0d2cd469d1bc616ffaca539/webassembly/app.html#L102-L110)
 - so I can only watch for when these `axis, grid, fxaa, tone, ssao, ambient` url-params are set to false, in which case I toggle them a second time before calling a final `Module.engineInstance.getWindow().render();` just before returnging from method `load_from_url()`

The current implementation does work under these assumptions from my testing. 
I think this PR can be merged, and the code could then be updated to account for a new `set_boolean` function + binding, or even higher-level `set_as_string` binding, once either is available.

Reply:
Alright, this is correct, I don't usually change hash after page was loaded but you're right. I'll include your modifications.